src/core/qgsarchive.cpp

```diff
@@ -95,14 +95,27 @@ void QgsArchive::addFile( const QString &file )
   mFiles.append( file );
 }
 
+bool QgsArchive::removeFile( const QString &file )
+{
+  bool rc = false;
+
+  if ( !file.isEmpty() && QFile::exists( file ) )
+    rc = QFile::remove( file );
+
+  if ( rc )
```

Comment:
> What is this check here for?

I remove the file from the archive only if it's well removed from the filesystem but it's indeed maybe not a good idea... Because, as you said, I should have a "recovery scenario" to, for example, retry later in case of other process blocking the deletion... 

I should probably ignore that case considering that the aim is just to NOT zip the file the next time.

And by the way, I should check that the file is well contained within the archive before removing it...